### ISSUE TITLE
How to change my jax code when I have a BatchNorm Layer?

### ISSUE THREAD

**Issue by Waterkin** [deepmind/dm-haiku#405]
Due to parameter 'is_training' of BatchNorm __call__() , I have to change my jax code
How to deal with` Jax.value_and_loss(params, x, y), model.apply(params, rng, x)` after adding a BatchNorm Layer?
Where should I add this paramter 'is_training'?
Thanks in advance!

**Comment by tomhennigan** [comment]
Hi @Waterkin , we have a fully worked example of training with `BatchNorm` in our imagenet example. You can see the call to grad here: https://github.com/deepmind/dm-haiku/blob/main/examples/imagenet/train.py#L154-L155

You can see we use `has_aux=True` since we need to return the updated batch norm states as part of the computation of the gradients. We include the loss in this auxiliary output, we do this (instead of using `value_and_grad`) since we scale our loss for the gradient computation and we want to return the unscaled loss for logging.

Here is a standalone example of BatchNorm + value_and_grad: https://colab.research.google.com/gist/tomhennigan/ae159586d26089100162b49bd3909f66/example-of-value_and_grad-with-batchnorm.ipynb

**Comment by uduse** [comment]
@tomhennigan I have a similar issue here (#471) that uses `BatchNorm` and returns different values for `is_training = True` and `is_training = False`. This is not normal right?
